### PR TITLE
Setup UI layer for new MySite screen

### LINKED ISSUE #13341: Setup UI layer for My Site
Setup Adapter, Base ViewHolder and a parent UI model for the improved my site. These should all be extensible. The UI model should have an enum type. There doesn't need to be an actual view holder implementation yet.

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #13341 and #13342 

I've used this PR to handle 2 issues (cause they seemed really related). I've created an Adapter, a DiffCallback, ViewModel, put everything into the Fragment so it should be ready for implementation of all the basic list items. 

To test:
- Turn on the MySiteImprovements flag (App settings/Test feature configuration)
- Restart the app
- Notice the empty My Site fragment
- Notice nothing is crashing

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt [hunk 1]
```diff
@@ -1,9 +1,78 @@
...
+    }
+
+    private fun loadData(items: List<MySiteItem>) {
+        if (recycler_view.adapter == null) {
+            recycler_view.adapter = MySiteAdapter()
+        }
+        val adapter = recycler_view.adapter as MySiteAdapter
+        val recyclerViewState = recycler_view.layoutManager?.onSaveInstanceState()
+        adapter.loadData(items)
+        recycler_view.layoutManager?.onRestoreInstanceState(recyclerViewState)
```
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt]: I'm not sure I understand why we need to call `onSaveInstanceState()` and `onRestoreInstanceState()` here.

**Review by renanferrari — COMMENTED** [review]
Thanks for the changes! I tested the app with the flag enabled and can confirm everything works as expected. Code also looks good, but I left a question below.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt]: hmm, good question. Generally it's there to keep scroll position after data changes but I think in this case it's not necessary (cause we don't expect massive changes to the data). I'll remove it.

**Review by renanferrari — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**